### PR TITLE
Add AllowRepeats to SBCommandInterpreterRunOptions.

### PR DESCRIPTION
**Pull request by jimingham**
This is useful if you have a transcript of a user session and want to rerun those commands with RunCommandInterpreter.  The same functionality is also useful in testing.

I'm adding it primarily for the second reason.  In a subsequent patch, I'm adding the ability to Python based commands to provide their "auto-repeat" command.  Among other things, that will allow potentially state destroying user commands to prevent auto-repeat.  Testing this with Shell or pexpect tests is not nearly as accurate or convenient as using RunCommandInterpreter, but to use that I need to allow auto-repeat.

I think for consistency's sake, having interactive sessions always do auto-repeats is the right choice, though that's a lightly held opinion...

### CODE REVIEW

File: lldb/include/lldb/Interpreter/CommandInterpreter.h [hunk 1]
```diff
@@ -93,15 +93,20 @@ class CommandInterpreterRunOptions {
...
+  /// \param[in] process_repeats
+  ///    If \b true then process empty lines as repeat commands even if the
+  ///    interpreter is non-interactive.
   CommandInterpreterRunOptions(LazyBool stop_on_continue,
                                LazyBool stop_on_error, LazyBool stop_on_crash,
                                LazyBool echo_commands, LazyBool echo_comments,
                                LazyBool print_results, LazyBool print_errors,
-                               LazyBool add_to_history)
+                               LazyBool add_to_history,
+                               LazyBool process_repeats)
```
Comment (medismailben) [inline lldb/include/lldb/Interpreter/CommandInterpreter.h:104]: nit: not sure what you mean by `process_repeats`, may be `command_repeats` would be make more sense here.

**Review by medismailben — APPROVED** [review]
LGTM with nit.

File: lldb/include/lldb/Interpreter/CommandInterpreter.h (hunk 1 above)
Comment (jimingham) [reply lldb/include/lldb/Interpreter/CommandInterpreter.h:104]: I was trying to stay in line with the other parameter names in this function.  So for instance, add_to_history says commands should be added to the history.  The option I'm adding is to tell RunCommandInterpreter to process any repeat commands in the input stream.

Comment (jimingham) [reply lldb/include/lldb/Interpreter/CommandInterpreter.h:104]: Internally in lldb we call the handling of newlines in the interactive interpreter as "repeat commands", and at the the CommandInterpreterRunOptions setting option was "AllowRepeats".

Comment (medismailben) [reply lldb/include/lldb/Interpreter/CommandInterpreter.h:104]: > tell RunCommandInterpreter to process any repeat commands

Oh, `process` is a verb here --' when I read it I thought you were talking about the `Process` class which confused me a little ^^

Comment (jimingham) [reply lldb/include/lldb/Interpreter/CommandInterpreter.h]: Sure.

File: lldb/include/lldb/API/SBCommandInterpreterRunOptions.h [hunk 2]
```diff
@@ -72,6 +72,14 @@ class LLDB_API SBCommandInterpreterRunOptions {
 
   void SetSpawnThread(bool);
 
+  bool GetAllowRepeats() const;
+
+  // By default, RunCommandInterpreter will discard repeats if the
+  // IOHandler being used is not interactive.  Setting AllowRepeats to true
+  // will override this behavior and always process empty lines in the input
+  // as a repeat command.
```
Comment (JDevlieghere) [inline lldb/include/lldb/API/SBCommandInterpreterRunOptions.h:80]: These should be `///` so they get picked up by Doxygen.

File: lldb/include/lldb/Interpreter/CommandInterpreter.h [hunk 3]
```diff
@@ -182,6 +187,11 @@ class CommandInterpreterRunOptions {
   void SetSpawnThread(bool spawn_thread) {
     m_spawn_thread = spawn_thread ? eLazyBoolYes : eLazyBoolNo;
   }
+  bool GetAllowRepeats() const { return DefaultToNo(m_allow_repeats); }
```
Comment (JDevlieghere) [inline lldb/include/lldb/Interpreter/CommandInterpreter.h:190]: Nit: newline between `SetSpawnThread` and `GetAllowRepeats`?

File: lldb/source/Interpreter/CommandInterpreter.cpp [hunk 4]
```diff
@@ -3129,14 +3130,19 @@ void CommandInterpreter::IOHandlerInputComplete(IOHandler &io_handler,
       return;
 
   const bool is_interactive = io_handler.GetIsInteractive();
-  if (!is_interactive) {
+  bool allow_repeats =
```
Comment (JDevlieghere) [inline lldb/source/Interpreter/CommandInterpreter.cpp:3133]: nit: `const bool` for consistency with the line above.

**Review by JDevlieghere — APPROVED** [review]
LGMT modulo nits.

File: lldb/include/lldb/Interpreter/CommandInterpreter.h (hunk 3 above)
Comment (jimingham) [reply lldb/include/lldb/Interpreter/CommandInterpreter.h:191]: done

**Pull request merged**